src/request_handler.rs

```diff
@@ -59,6 +69,43 @@ impl RequestHandler {
 
     Ok(Response::new(Body::from(body.into_string())))
   }
+
+  async fn serve_file(&self, file_path: &FilePath) -> Result<Response<Body>> {
+    // todo: stream files
+    let file_contents =
+      tokio::fs::read(dbg!(self.working_directory.join("www").join(file_path))).await?;
+    Ok(Response::new(Body::from(file_contents)))
+  }
+}
+
+#[derive(Debug)]
```

Comment:
I noticed that a bad file path could escape the `www` directory, so I wrote `FilePath` before I noticed `disallow_access_outside_of_www`, so maybe you wanted to leave it for later.

This wrapper sanitizes file paths. It uses `Path::components` and complains if there are anything but "normal" path components.

Reply:
This could also just be a function, instead of a type:

```
fn file_path_from_uri(uri: &Uri) -> &str
```